src/pool-cl/libraries/CLPool.sol

```diff
@@ -110,6 +113,8 @@ library CLPool {
         Tick.checkTicks(params.tickLower, params.tickUpper);
 
         (uint256 feesOwed0, uint256 feesOwed1) = _updatePosition(self, params, _slot0.tick);
+        // @dev track swap fees for off-chain farming
+        emit SwapFees(feesOwed0, feesOwed1);
```

Comment:
I guess the position info is also needed ? Otherwise how do u know which pool / position is generating fee ?

Reply:
I think I should emit this event in nfp instead

Reply:
move to nfp, plz see https://github.com/pancakeswap/pancake-v4-periphery/pull/15